### ISSUE TITLE
Empty line in codeblock breaks formatting

### ISSUE THREAD

**Issue by ghost** [Keats/gutenberg#2348]
# Bug Report

In a "backticks" codeblock without line numbers, an empty line will insert `<p></p>` tags around the code with broken nesting like so:

```html
<pre><code><p></code></pre></p>
```

## Environment

Zola version: 0.17.2

## Expected Behavior

An empty line should not produce a malformed code block.

## Current Behavior

When using a codeblock defined using backtics, an empty line will generate a paragraph inside the codeblock with broken nesting:

````md
```
backticks codeblock

Empty lines break things
```
````

```html
<pre class="z-code"><code><span class="z-text z-plain">backticks codeblock
<p>Empty lines break things
</span></code></pre></p>
```

By contrast, a codeblock using empty spaces will generate a `<pre><code></code></pre>` block.

```md
    4-space codeblock
    
    Empty Lines work
```

```html
<pre class="z-code"><code><span class="z-text z-plain">4-space codeblock

Empty lines work
</span></code></pre>
```
### linenos

Blocks with `linenos` are not affected.

````html
<pre data-linenos class="z-code"><code><table><tbody><tr><td>1</td><td><span class="z-text z-plain">backticks codeblock
</td></tr><tr><td>2</td><td>
</td></tr><tr><td>3</td><td>Empty lines break things
</td></tr></span></tbody></table></code></pre>
````

## Step to reproduce

`zola build --drafts` a site with the following minimal reproducible example:

````markdown
---
title: "Codeblock Test"
description: "A page for testing codeblock generation"
date: "2023-11-01"
draft: true
---

```
backticks codeblock

Empty lines break things
```

    4-space codeblock
    
    Empty Lines work
````

**Comment by Keats** [comment]
It looks like they generate the same thing on the `next` branch?

```
<pre style="background-color:#383838;color:#e6e1dc;"><code><span>backticks codeblock
</span><span>
</span><span>Empty lines break things
</span></code></pre>
<pre style="background-color:#383838;color:#e6e1dc;"><code><span>4-space codeblock
</span><span>
</span><span>Empty Lines work
</span></code></pre>
```

**Comment by ghost** [comment]
@Keats I will try that out when I get the chance! I also found a workaround in the meantime, placing a single space in any empty line prevents the `<p>` tag insertion.

**Comment by Raymi306** [comment]
I may have a related bug on v0.18, I have a random newline added to the beginning and end of every codeblock that I am putting a shortcode inside of.  I cannot strip these newlines with filters as best as I can tell.

```html
<pre data-lang="py" style="background-color:#fdf6e3;color:#657b83;" class="language-py "><code class="language-py" data-lang="py"><span>
</span><span style="color:#859900;">print</span><span>(</span><span style="color:#839496;">'</span><span style="color:#2aa198;">Hello world</span><span style="color:#839496;">'</span><span>)
</span><span>
</span></code></pre>
```

I may make a new issue after I rest some.

**Comment by ghost** [comment]
Confirmed fix on v0.18.0
